src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -218,22 +226,24 @@ def serialize(self, obj, attrs, user, **kwargs):
 
 class KeyTransactionTeamSerializer(Serializer):
     def get_attrs(self, item_list, user, **kwargs):
-        counts = (
-            TeamKeyTransaction.objects.values("team_id")
-            .filter(team__in=[item.id for item in item_list])
-            .annotate(count=Count("team_id"))
-        )
-        count_map = {}
-        for count in counts:
-            count_map[count["team_id"]] = count["count"]
+        team_key_transactions = TeamKeyTransaction.objects.filter(
+            team__in=[item.id for item in item_list]
+        ).order_by("transaction", "project_id")
+
+        attrs = defaultdict(lambda: {"key_transactions": []})
+
+        for kt in team_key_transactions:
+            attrs[kt.team]["key_transactions"].append(
```

Comment:
I wonder if we should add a `count` here so we can filter some of the transactions we return out based on project params?

Reply:
1. adding a `count` here doesnt provide any additional information that the list doesn't already provide
2. we can't filter based on the project params because the 100 key transactions per team limit applies across projects, so if we filter it out here, the frontend won't be able to show the appropriate error messages.

Reply:
Hm so what I'm thinking is we still run the Query with all projects. but in the response we filter out the key transactions that aren't part of the project list. This means that:
1. the `count` would provide additional information since we no longer have all the key transactions in the response
2. we use the count to show the appropriate error message instead

Reply:
That makes sense 👍 updating with this suggestion

Reply:
Updated in 261dc93